DQM/BeamMonitor/plugins/OnlineBeamMonitor.cc

```diff
@@ -128,6 +128,49 @@ void OnlineBeamMonitor::bookHistograms(DQMStore::IBooker& ibooker,
   bsChoice_->setAxisTitle("Choice", 2);
 }
 
+//----------------------------------------------------------------------------------------------------------------------
+// Handle exceptions for the schema evolution of the BeamSpotOnline CondFormat
+
+// Slightly better error handler
+static void print_error(const std::exception& e) { edm::LogError("BeamSpotOnlineParameters") << e.what() << '\n'; }
+
+// Method to catch exceptions
+template <typename T, class Except, class Func, class Response>
+T try_(Func f, Response r) {
+  try {
+    LogDebug("BeamSpotOnlineParameters") << "I have tried" << std::endl;
+    return f();
+  } catch (Except& e) {
+    LogDebug("BeamSpotOnlineParameters") << "I have caught!" << std::endl;
+    r(e);
+    return static_cast<T>("-999");
+  }
+}
+
+// Enum the BS string parameters
+enum BSparameters {
+  startTime = 0,  // 0 additional std::string parameters
+  endTime = 1,    // 1
+  lumiRange = 2,  // 2
+  END_OF_TYPES = 3,
+};
+
+// Functor
+std::function<std::string(BSparameters, BeamSpotOnlineObjects)> myStringFunctor = [](BSparameters my_param,
```

Comment:
I wonder if
```suggestion
static const std::function<std::string(BSparameters, BeamSpotOnlineObjects)> myStringFunctor = [](BSparameters my_param,
```
could shut off the warning of the static analyzer : 
```
Non-const variable 'std::function<std::string (BSparameters, BeamSpotOnlineObjects)> myStringFunctor' is static and might be thread-unsafe
```

And maybe another question could be: is worth doing it?

Reply:
Ciao Andrea,
I can give it a try, but I'm currently not able to reproduce the warning locally.
I tried running on lxplus8 the command from the [PR tests](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-80b02d/30539/llvm-analysis/):
```
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02771/share/lcg/SCRAMV1/V3_00_58/bin/scram build -f compile COMPILER=llvm-analyzer SCRAM_IGNORE_MISSING_COMPILERS=yes
```
But I only get 3 warnings instead of the 62 reported in the [SA checks](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-80b02d/30539/llvm-analysis/)...
Maybe you can tell me if i'm doing something wrong or suggest me how to reproduce it exactly?

Reply:
> Maybe you can tell me if i'm doing something wrong or suggest me how to reproduce it exactly?

Not really, I normally trust on the bot tests
Making it `static const` should work to remove the warning; from the practical point of view probably nothing will change though

Reply:
Thanks for the merge Andrea!
Indeed there are a few SA warnings in this package that can be followed up later on.

Reply:
> But I only get 3 warnings instead of the 62 reported in the [SA checks](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-80b02d/30539/llvm-analysis/)...

@smuzaffar, since I often incur in the same issue, for my own edification, could you clarify how to reproduce locally the SA checks output?

Reply:
Self-answering:
I stumbled upon this [SCRAM guide](https://readthedocs.org/projects/scram/downloads/pdf/latest/) that reports the command `scram b checker` which `"Run LLVM static checkers"`.
And indeed it runs exactly as the SA checks and reports the same warnings (and even a few more...).

FYI @mmusich